### PR TITLE
Add Mistral integration page

### PR DESCRIPTION
**Pull request by annthurium**
Now that we have a Mixtral API key, tested these code samples! You can try it yourself [in this colab](https://colab.research.google.com/drive/1LNFwFcQBwEUrY11XZolltFhLWSQ4oTUO#scrollTo=Rbo1rzxvnxTA).

I tried to include an example of using the embedding models in a pipeline. However, it looks like the `OpenAIDocumentEmbedder` doesn't support the Mistral models out of the box, unless I'm missing something. You can see more in the Colab. Do you have recommendations on how to adapt that or should I just leave as is? Thank you.

### CODE REVIEW

File: integrations/mistral.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+repo: https://github.com/deepset-ai/haystack
+type: Model Provider
+report_issue: report_issue: https://github.com/deepset-ai/haystack/issues
+logo: /logos/mistral.png
+version: Haystack 2.0
+toc: true
+---
+
+This page demonstrates how to use the [OpenAIChatGenerator](https://docs.haystack.deepset.ai/v2.0/docs/openaichatgenerator) and [OpenAITextEmbedder](https://docs.haystack.deepset.ai/v2.0/docs/openaitextembedder) within Haystack to make use of Mistral models.
+
```
Comment (TuanaCelik) [inline integrations/mistral.md:21]: Can we add a single line here that explains what's going on here something like:
"We're able to use the OpenAI generators for Mistral as they use the same protocol".. And then add a line explaining that this is possible by changing the api_base_url

File: integrations/mistral.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+### Installation
+
+```bash
+pip install haystack-ai
+```
+
+### Usage
+
+To use a Mistral LLM:
```
Comment (TuanaCelik) [inline integrations/mistral.md:39]:
```suggestion
#### Use Mistral Generative Models
```

File: integrations/mistral.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    streaming_callback=lambda chunk: print(chunk.content, end="", flush=True)
+)
+
+response = client.run(
+    messages=[ChatMessage.from_user("What is the best French cheese?")]
+)
+print(response)
+```
+
+To use a Mistal embedding model:
```
Comment (TuanaCelik) [inline integrations/mistral.md:82]:
```suggestion
#### Use a Mistral Embedding Models
```

**Review by TuanaCelik — COMMENTED** [review]
Some comments, but one last question: are we not able to use the OpenAIGenerator too? Is it only the chat generator?

**Comment by annthurium** [comment]
@TuanaCelik re the `OpenAIGenerator`: according to [Mistral's docs](https://docs.mistral.ai/api/) the only exposed endpoints their platform provides are
 1) chat completion
2) creating text embeddings
3) listing available models.

So I assumed we can't use the `OpenAIGenerator` but I can try anyway if you'd like.

**Comment by TuanaCelik** [comment]
@annthurium I guess it doesn't really matter then. Sounds like even if we _can_, using and telling people to use the ChatGenerator would be more correct. But let's ask @anakin87 just in case

**Comment by anakin87** [comment]
Mistral API expects `messages`, so ChatGenerator is the right abstraction in this case.
(Generators do not pass messages to the API, so I would expect a failure).

_Unrelated_: Embedders work as expected (wrt https://github.com/deepset-ai/haystack-integrations/pull/125#issue-2082833272)
`document_embedder = OpenAIDocumentEmbedder(api_key=userdata.get("MISTRAL_API_KEY"), model='mistral-embed', api_base_url="https://api.mistral.ai/v1")`

**Comment by annthurium** [comment]
@anakin87 when I try to use the `OpenAIDocumentEmbedder` in a Haystack pipeline I run into the following error:
```python
from haystack import Document
from haystack import Pipeline
from haystack.document_stores.in_memory import InMemoryDocumentStore
from haystack.components.embedders import OpenAITextEmbedder, OpenAIDocumentEmbedder
from haystack.components.retrievers.in_memory import InMemoryEmbeddingRetriever

document_store = InMemoryDocumentStore(embedding_similarity_function="cosine")

documents = [Document(content="My name is Wolfgang and I live in Berlin"),
             Document(content="I saw a black horse running"),
             Document(content="Germany has many big cities")]

document_embedder = OpenAIDocumentEmbedder(api_key=userdata.get("OPENAI_API_KEY"), model='mistral-embed')
documents_with_embeddings = document_embedder.run(documents)['documents']
```
```
[/usr/local/lib/python3.10/dist-packages/openai/_base_client.py](https://localhost:8080/#) in _request(self, cast_to, options, remaining_retries, stream, stream_cls)
    958 
    959             log.debug("Re-raising status error")
--> 960             raise self._make_status_error_from_response(err.response) from None
    961 
    962         return self._process_response(

NotFoundError: Error code: 404 - {'error': {'message': 'The model `mistral-embed` does not exist', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}
```

**Comment by annthurium** [comment]
actually - when I pass in the API_BASE_URL I get an authentication error :-( 

```python
from haystack import Document
from haystack import Pipeline
from haystack.document_stores.in_memory import InMemoryDocumentStore
from haystack.components.embedders import OpenAITextEmbedder, OpenAIDocumentEmbedder
from haystack.components.retrievers.in_memory import InMemoryEmbeddingRetriever

document_store = InMemoryDocumentStore(embedding_similarity_function="cosine")

documents = [Document(content="My name is Wolfgang and I live in Berlin"),
             Document(content="I saw a black horse running"),
             Document(content="Germany has many big cities")]

document_embedder = OpenAIDocumentEmbedder(api_key=userdata.get("OPENAI_API_KEY"), model='mistral-embed', api_base_url="https://api.mistral.ai/v1")
documents_with_embeddings = document_embedder.run(documents)['documents']
```
```
AuthenticationError                       Traceback (most recent call last)
[<ipython-input-4-d24898994ea8>](https://localhost:8080/#) in <cell line: 14>()
     12 
     13 document_embedder = OpenAIDocumentEmbedder(api_key=userdata.get("OPENAI_API_KEY"), model='mistral-embed', api_base_url="https://api.mistral.ai/v1")
---> 14 documents_with_embeddings = document_embedder.run(documents)['documents']
     15 document_store.write_documents(documents)
     16
    959             log.debug("Re-raising status error")
--> 960             raise self._make_status_error_from_response(err.response) from None
    961 
    962         return self._process_response(

AuthenticationError: Error code: 401 - {'message': 'Unauthorized', 'request_id': '99b1cd757ca42d71e7c7182e8a10f20d'}
```

**Comment by anakin87** [comment]
I think you should pass the `MISTRAL_API_KEY`

**Comment by annthurium** [comment]
@anakin87 oh thanks, that was silly of me! Cool, the pipeline example works now!! I'll clean it up and add it in here.

**Review by TuanaCelik — APPROVED** [review]

**Pull request merged**